### PR TITLE
CNV-37570: Doc: use pipelines to manage VMs and boot sources

### PR DESCRIPTION
**Pull request by sbeskin-redhat**
Resolves: https://issues.redhat.com/browse/CNV-37570

OCP 4.16
CNV 4.16

Preview:
https://77640--ocpdocs-pr.netlify.app/openshift-dedicated/latest/virt/virtual_machines/creating_vms_rh/virt-creating-vms-from-instance-types.html#virt-creating-vm-instancetype_virt-creating-vms-from-instance-types

### CODE REVIEW

**Comment by yfrimanm** [comment]
Hi @sbeskin-redhat 
In the second bullet: 

> Click Add volume to upload a new volume or use an existing persistent volume claim (PVC), volume snapshot, or data source. Then click Save.

I think we can simplify it and say “Click Add volume to upload a new volume or use an existing persistent volume claim (PVC). Then click Save.”
@dominikholler should we specify that this must be a DataSource?

I also think that this paragraph is too complexed:

> If bootable volumes for a certain operating system (Windows, RHEL, or Linux) are missing in the list, this is indicated by the operating system logo that appears under the list. The logo is followed by a suggestion to add a volume for the missing operating system by clicking the Add volume link. In addition, there is a link to the Create a Windows boot source quick start. The same link appears in a popover if you hover the pointer over the question mark icon next to the Select volume to boot from line.

Maybe we can simplify it by saying something like "Logos of operating systems that are not available in the cluster are shown at the bottom of the list to hint the user that they can be added."

**Comment by dominikholler** [comment]
> I think we can simplify it and say “Click Add volume to upload a new volume or use an existing persistent volume claim (PVC). Then click Save.”

The current doc matches the first line in the `Add volume` dialog. Technically correct would be 
"Click 'Add volume' to upload a new volume or use an existing volume (PVC), snapshot or a containerdisk. Then click Save."
This might be adjusted also in the UI.

> @dominikholler should we specify that this must be a DataSource?

The "Add volume" UI will ensure that a DataSource will be created, like expected, no need to explain this to the user.

**Comment by yfrimanm** [comment]
@dominikholler I am aware this matches the current text we currently have in the `Add volume` dialog. 
We were thinking to simplify it and make it more user-friendly in the UI as well. WDYT?
cc @ronensdeor

**Comment by dominikholler** [comment]
>  I am aware this matches the current text we currently have in the Add volume dialog.
We were thinking to simplify it and make it more user-friendly in the UI as well. WDYT?

I like to support this idea. Would a generalization/abstraction instead of a list of examples be simpler?

“Click `Add volume` to use an already existing disk image”

**Comment by sbeskin-redhat** [comment]
> +1 on generalizing it. I think we need to make it clear in the sentence that they have 2 options. Maybe something like: “Click `Add volume` to upload a **new** volume or use an **existing** disk image from the volumes list. Then click Save.”

@yfrimanm Which list are you referring to here? Is it inside the 'Add volume' dialog?

**Comment by yfrimanm** [comment]
I meant the "Volumes list" that is shown in the 1st step "Select volume to boot from", but maybe I went too "high-level" here.

**Comment by sbeskin-redhat** [comment]
> I meant the "Volumes list" that is shown in the 1st step "Select volume to boot from", but maybe I went too "high-level" here.

@yfrimanm I am a bit confused here. The user has 2 options - to choose a volume from the list displayed in the instanceTypes tab or to click 'Add volume'. If he clicks 'Add volume', the first line of the dialog says: Upload a new volume or use an existing PVC, snapshot or DataSource.  Where are those 'existing' volumes appear? Is there an additional list somewhere?

**Comment by yfrimanm** [comment]
@sbeskin-redhat from the dialog users can select the source type: they can upload a new volume, they can use existing volume or volume snapshot and they can also import from URL or registry (but I'm not sure we need to mention the last option of the import).
The volumes list is under the "Show all" button, but they can also use the dialog as I mentioned above.

**Comment by ronensdeor** [comment]
'existing volumes' means volumes (or volume snapshots) that are on the
cluster, or volumes that reside in a registry and the user needs to set the
url to it.
It will not show volumes that are already in the list

**Comment by sbeskin-redhat** [comment]
@ronensdeor Oh, I see, thank you.

File: modules/virt-creating-vm-instancetype.adoc [hunk 1]
```diff
@@ -39,19 +39,26 @@ endif::[]
...
-* Click *Add volume* to upload a new volume or use an existing persistent volume claim (PVC), volume snapshot, or data source. Then click *Save*.
+Optional: Click the star icon to designate a bootable volume as a favorite. Starred bootable volumes appear first in the volume list.
 
+* Click *Add volume* to upload a new volume or to use an existing persistent volume claim (PVC), a volume snapshot, or a `containerDisk` volume. Click *Save*.
++
+Logos of operating systems that are not available in the cluster are shown at the bottom of the list. You can add a volume for the required operating system by clicking the *Add volume* link.
++
+In addition, there is a link to the *Create a Windows boot source* quick start. The same link appears in a popover if you hover the pointer over the question mark icon next to the _Select volume to boot from_ line.
++
+Immediately after you install the environment or when the environment is disconnected, the list of volumes to boot from is empty. In that case, three operating system logos are displayed (Windows, RHEL, and Linux), and you can add a new volume that meets you requirements by clicking the *Add volume* button.
```
Comment (jldohmann) [inline modules/virt-creating-vm-instancetype.adoc:61]:
```suggestion
Immediately after you install the environment or when the environment is disconnected, the list of volumes to boot from is empty. In that case, three operating system logos are displayed: Windows, {op-system-base}, and Linux. You can add a new volume that meets you requirements by clicking the *Add volume* button.
```

**Review by jldohmann — COMMENTED** [review]
It's looking good! just some minor comments and suggestions below

File: modules/virt-creating-vm-instancetype.adoc [hunk 2]
```diff
@@ -39,19 +39,26 @@ endif::[]
...
 +
 [NOTE]
 ====
 The bootable volume table lists only those volumes in the `openshift-virtualization-os-images` namespace that have the `instancetype.kubevirt.io/default-preference` label.
 ====
 +
-** Optional: Click the star icon to designate a bootable volume as a favorite. Starred bootable volumes appear first in the volume list.
-
-* Click *Add volume* to upload a new volume or use an existing persistent volume claim (PVC), volume snapshot, or data source. Then click *Save*.
+Optional: Click the star icon to designate a bootable volume as a favorite. Starred bootable volumes appear first in the volume list.
```
[the remark this answers is not in the document]
Comment (sbeskin-redhat) [reply modules/virt-creating-vm-instancetype.adoc:53]: @jldohmann 
Hi,
Do you mean making it a sub-bullet of the previous bullet? But why?

[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/virt-creating-vm-instancetype.adoc:53]: yes, from the ISG: "If a step is optional, begin the step with the word _Optional_", which to me implies that it should be a separate step

[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/virt-creating-vm-instancetype.adoc:53]: > It belongs to the first option. If you select a volume from the list, you can mark it as favorite.

sounds like a substep to me!

[the remark this answers is not in the document]
Comment (sbeskin-redhat) [reply modules/virt-creating-vm-instancetype.adoc:53]: @jldohmann 
I see your point but a single sub-bullet looks strange, don't you think?

[the remark this answers is not in the document]
Comment (jldohmann) [reply modules/virt-creating-vm-instancetype.adoc:53]: we have single-step procedures in the documentation and the occasional single sub-bullet so maybe i'm used to how it looks 😄 but ultimately i'm just reviewing against our guidelines

[the remark this answers is not in the document]
Comment (sbeskin-redhat) [reply modules/virt-creating-vm-instancetype.adoc]: @jldohmann OK, I've made it a sub-bullet.

**Comment by sbeskin-redhat** [comment]
@jldohmann Thank you for the review!

**Review by sheriff-rh — APPROVED** [review]
Changes LGTM, merging to 4.16+.

**Comment by sbeskin-redhat** [comment]
@sheriff-rh  Thank you!

**Pull request merged**